runtime/doc/diagnostic.txt

```diff
@@ -39,16 +39,18 @@ modify the diagnostics for a buffer (e.g. |vim.diagnostic.set()|) then it
 requires a namespace.
 
 							*diagnostic-structure*
-A diagnostic is a Lua table with the following keys:
+A diagnostic is a Lua table with the following keys. Required keys are
+indicated with (*):
 
 	bufnr: Buffer number
-	lnum: The starting line of the diagnostic
+	lnum(*): The starting line of the diagnostic
 	end_lnum: The final line of the diagnostic
-	col: The starting column of the diagnostic
+	col(*): The starting column of the diagnostic
 	end_col: The final column of the diagnostic
 	severity: The severity of the diagnostic |vim.diagnostic.severity|
-	message: The diagnostic text
+	message(*): The diagnostic text
```

Comment:
Something to note: `message` is supposed to be a required field, but there's no `assert` that's checking for it like there is with `lnum` and `col`: https://github.com/neovim/neovim/blob/4c24fcb91f7740410b94638b3156ba20647df579/runtime/lua/vim/diagnostic.lua#L272-L276

So I can `vim.diagnostic.set()` a diagnostic without a message, but when I try to `vim.diagnostic.open_float()` it, the call fails. If that is indeed unintended, then I can open an issue for it. It's technically a breaking change so I'm not sure if I should address it in this PR

Reply:
Hmm yes I think we should assert for `diagnostic.message` as well. I don't consider that a breaking change however, as the diagnostics are already assuming that `message` exists, so omitting it already causes errors (as you've noticed), we'd just be making the error more explicit.